### PR TITLE
[dispatcher] allow receive UDP events from remote nodes

### PR DESCRIPTION
**Pull request by simonlingoogle**
Originally OTNS listens at 127.0.0.1:9000 for UDP events so it must run locally with simulation nodes. 
This PR enhances OTNS to allow it to receive and send UDP events from remote nodes.

The source address of a received UDP event will be used as the peer address for that node.

### CODE REVIEW

**Comment by jwhui** [comment]
@simonlingoogle , can you help resolve conflicts?

File: dispatcher/dispatcher.go [hunk 1]
```diff
@@ -129,7 +129,9 @@ type Dispatcher struct {
 }
 
 func NewDispatcher(ctx *progctx.ProgCtx, cfg *Config, cbHandler CallbackHandler) *Dispatcher {
-	udpAddr, err := net.ResolveUDPAddr("udp4", ":9000")
+	simplelogger.AssertTrue(!cfg.Real || cfg.Speed == 1)
+
+  udpAddr, err := net.ResolveUDPAddr("udp4", ":9000")
```
Comment (ctan-g) [inline dispatcher/dispatcher.go:134]: Indentation seems wrong

Comment (simonlingoogle) [reply dispatcher/dispatcher.go]: Thanks

Comment (simonlingoogle) [reply dispatcher/dispatcher.go]: resolve

**Pull request merged**